### PR TITLE
Fixes MAISTRA-1226: fix for ScopedEVP_MD_CTX used in UtilityImpl::verifySignature

### PR DESCRIPTION
**Pull request by dmitri-d**
This is the first part of a fix for https://issues.redhat.com/browse/MAISTRA-1226. The second part is here: https://github.com/Maistra/envoy/pull/6

### CODE REVIEW

**Review by brian-avery — APPROVED** [review]

**Comment by dmitri-d** [comment]
@dgn: ping

File: bssl_wrapper/bssl_wrapper.h [hunk 1]
```diff
@@ -45,11 +45,11 @@ class StackAllocated {
 
   void Reset() {
     cleanup(ctx_);
-    init(ctx_);
+    init();
```
Comment (jwendell) [inline bssl_wrapper/bssl_wrapper.h]: don't you need to do `ctx_ = init()` here?

Comment (dmitri-d) [reply bssl_wrapper/bssl_wrapper.h]: No, pls. see https://github.com/Maistra/bssl_wrapper/pull/3/files#diff-1c12f051fb284995269056bf27f1382aR30

Comment (dmitri-d) [reply bssl_wrapper/bssl_wrapper.h]: Hmm, let me have another look...

Comment (dmitri-d) [reply bssl_wrapper/bssl_wrapper.h]: Yep, you are right. Fixed, good catch.

**Review by jwendell — APPROVED** [review]

**Pull request merged**